src/pointer-analysis/value_set.cpp

```diff
@@ -1220,7 +1221,8 @@ void value_sett::assign(
   std::cout << "--------------------------------------------\n";
   output(ns, std::cout);
 #endif
-
+  PRECONDITION(is_l1_renamed(lhs));
```

Comment:
This is only true when used in symex-- otherwise we expect to operate on plain un-renamed expressions. Therefore these invariants should move to `symex_dereference` or to some wrapper function in `goto_statet`.

Reply:
I defined a wrapper class for `value_sett` (a wrapper function wouldn't ensure we didn't continue to use the unchecked one)